code/modules/robotics/bot/guardbot.dm

```diff
@@ -826,16 +826,17 @@
 		var/long_day = "[pick("Hooh", "Yeah", "Yeesh", "Blimey")], [pick("wow,", "heh,", "huh,")] guess it's been a long [pick("day", "shift", "morning")][shiftTime > 6000 ? "." : " already!"]"
 		switch(trickery)
 			if ("togglelock")
-				speak("Sorry, only people authorized by Thinktronic Data Systems may access my controls and accessories.")
-				if (deceptioncheck_passed)
-					src.locked = !src.locked
-					SPAWN_DBG(2 SECONDS)
-						speak(its_the_rd)
-						speak(long_day)
-						speak("Okay, everything's [src.locked ? "locked" : "unlocked"] now!")
-					return 1
-				else
-					return 0
+				if(!ON_COOLDOWN(src, "speak cooldown", 2 SECOND))
+					speak("Sorry, only people authorized by Thinktronic Data Systems may access my controls and accessories.")
```

Comment:
why is this not inside the Else below

Reply:
Yeah the ifs here are not done correctly. Currently it'll show the distracted message and then lock/unlock anyway.